game-core/src/main/java/games/strategy/triplea/attachments/TerritoryAttachment.java

```diff
@@ -474,14 +480,29 @@ private void setWhenCapturedByGoesTo(final ArrayList<String> value) {
     m_whenCapturedByGoesTo = value;
   }
 
-  public ArrayList<String> getWhenCapturedByGoesTo() {
+  private ArrayList<String> getWhenCapturedByGoesTo() {
     return m_whenCapturedByGoesTo;
   }
 
   private void resetWhenCapturedByGoesTo() {
     m_whenCapturedByGoesTo = new ArrayList<>();
   }
 
+  public List<CaptureOwnershipChange> getCaptureOwnershipChanges() {
+    return m_whenCapturedByGoesTo.stream()
+        .map(this::parseCaptureOwnershipChange)
+        .collect(Collectors.toList());
+  }
+
+  private CaptureOwnershipChange parseCaptureOwnershipChange(final String encodedCaptureOwnershipChange) {
+    final String[] tokens = splitOnColon(encodedCaptureOwnershipChange);
+    assert tokens.length == 2;
```

Comment:
I was about to say that an assertion is probably not the best idea here and I'd have used an IllegalArgumentException instead, until I realised this is a private method and only ever called with checked content.
Although it seems storing Captureownershipchange objects in the first place would make savegames incompatible, it could work in theory because the generic type is not really being stored.
However I'd appreciate if a comment could be added to indicate this behaviour should be changed in the next incompatible release.

Reply:
> it could work in theory because the generic type is not really being stored

Not sure about that.  While the deserialization may succeed because there's no runtime  checking of the element types, when some downstream code goes to use the resulting `ArrayList<String>`, it will inevitably raise a `ClassCastException` when the elements of that list are accessed (e.g. to split on a colon).

> However I'd appreciate if a comment could be added to indicate this behaviour should be changed in the next incompatible release.

23de957fa